docking_server:
  ros__parameters:
    controller_frequency: 50.0
    initial_perception_timeout: 5.0
    wait_charge_timeout: 5.0
    dock_approach_timeout: 30.0
    undock_linear_tolerance: 0.1
    undock_angular_tolerance: 0.01
    max_retries: 5
    base_frame: "base_link"
    fixed_frame: "map"
    dock_backwards: false
    dock_prestaging_tolerance: 0.01
   # navigate_to_staging_pose: false

    # Types of docks
    dock_plugins: ['simple_charging_dock']
    simple_charging_dock:
      plugin: 'opennav_docking::SimpleNonChargingDock'
      docking_threshold: 0.01
      staging_x_offset: -0.65
      use_external_detection_pose: true
      use_battery_status: false # true
      use_stall_detection: false # true

      external_detection_timeout: 30.0
      external_detection_translation_x: -0.47
      external_detection_translation_y: 0.0
      external_detection_rotation_roll: 0.0
      external_detection_rotation_pitch: 1.57
      external_detection_rotation_yaw: 0.0
      filter_coef: 0.2

    # Dock instances
    # The following example illustrates configuring dock instances.
    docks: ['home_dock']  # Input your docks here
    home_dock:
      type: 'simple_charging_dock'
      frame: map
      pose: [1.97, 3.355, 1.57]

    controller:
      k_phi: 3.5
      k_delta: 2.5
      v_linear_min: 0.01
      v_linear_max: 0.08
      use_collision_detection: false
      costmap_topic: "local_costmap/costmap_raw"
      footprint_topic: "local_costmap/published_footprint"
      transform_tolerance: 0.1
      projection_time: 10.0
      simulation_step: 0.05
      dock_collision_threshold: 0.2